src/librustc_trait_selection/traits/mod.rs

```diff
@@ -259,7 +259,7 @@ fn do_normalize_predicates<'tcx>(
                 return Err(ErrorReported);
             }
         };
-        if predicates.has_local_value() {
+        if predicates.needs_infer() {
             // FIXME: shouldn't we, you know, actually report an error here? or an ICE?
             Err(ErrorReported)
```

Comment:
cc @nikomatsakis This is a scary FIXME comments :open_mouth:. Should we add a `delay_span_bug`?

Reply:
added a `delay_span_bug`. Should be a relevant error msg afaik
> encountered inference variables after `fully_resolve`